### DISCUSSION TITLE
Does this engine support 3D Character imports + retargeting?

### DISCUSSION THREAD

**Discussion by MangoDragonHub** [ikemen-engine/Ikemen-GO#1612] [Q&A]
This is my first time hearing about Ikemen Go as a project considering I used to play alot of MUGEN as a kid. I am curious if this engine supports the importing of 3D Playable Characters and be able to transfer animation data between players via retargeting? I know this engine DOES support GLTF models for stages though.

**Comment by SuperFromND** [comment]
Currently we only support 3D models for stages. I don't think anyone's proposed a format for 3D characters yet, but it'd almost certainly require a fundamentally different approach to 2D characters, at least in regards to things like CLSNs.

**Comment by MangoDragonHub** [comment]
Why not GLTF as well? That's a readable and open format people can use. You don't need to make a custom FBX parser to read FBX files.

**Comment by MangoDragonHub** [comment]
But then again, you pretty much have to make a decent 3D Renderer for all this to work as well as keeping the 2D Engine. I tried myself following a 3D Java tutorial and there are a lot of components you have to keep in mind.

**Comment by SuperFromND** [comment]
Oh I should clarify, by format I meant *character* formats rather than file formats (we support GLTF for stages and that format is, tech-wise, entirely capable of storing a 3D character with an animatable rig and multiple animations; heck, we even added armature support a couple months back). By that I mean, stuff like defining hitboxes; trivial enough for SFF, not so much for a 3D model. Not to mention you'd have to standardize how animations would work since GLTF doesn't use the group/index system like SFF does (instead GLTF animations have full-on proper names and not just numeric IDs), and then things like palettes, certain state controllers, yada yada yada.

Not to say it's _impossible_ as in theory Ikemen is perfectly capable of rendering 3D characters, but nobody's really worked on laying out how that would actually *work* from a character-developement standpoint.

**Comment by MangoDragonHub** [comment]
Ah! I see what you mean, I forget that you have to have a file format that bundles all that data in while also still be readable if need to edit it. Luckily GLTF is a readable format to begin with, we can start from there with the vertex and skin data. When having to implement hitboxs/collision detection for a character it uses those points to toggle on and off when a player is strucken there. In the read file would be like some true or false thing. 
But yeah, it's going to take awhile to just fundementals of something big as this to figure out.

**Comment by MangoDragonHub** [comment]
I know some Youtubers like ThinMatrix in his Java games when writing a file format for his 3D Object it gets a reference to the 3D model data and add extra variables on top for the readable format he uses.
